File: app/src/main/java/com/example/test1/model/PostRepository.java
package com.example.test1.model;

import androidx.lifecycle.MutableLiveData;

import com.example.test1.Retrofit.APIClient;
import com.example.test1.Retrofit.APIService;
import com.example.test1.Util;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PostRepository {
  private ArrayList<Data> post = new ArrayList<>();
  private MutableLiveData<List<Data>> mutableLiveData = new MutableLiveData<>();

  public PostRepository() {
  }

  public MutableLiveData<List<Data>> getMutableLiveData() {

    final APIService userDataService = APIClient.getClient();

    Call<Post> call = userDataService.getPost();
    call.enqueue(new Callback<Post>() {
      @Override
      public void onResponse(Call<Post> call, Response<Post> response) {
        Post Post = response.body();
        if (Post != null && Post.getHits().getHits() != null) {
          post = (ArrayList<Data>) Util.convertData(Post.getHits().getHits());
          mutableLiveData.setValue(post);
        }
      }

      @Override
      public void onFailure(Call<Post> call, Throwable t) {
      }
    });

    return mutableLiveData;
  }
}

File: app/src/main/java/com/example/test1/model/PostRepository.kt
package com.example.test1.model

import android.app.Application
import android.os.AsyncTask
import androidx.lifecycle.MutableLiveData
import com.example.test1.Retrofit.APIClient
import com.example.test1.constant.Util.convertData
import com.example.test1.database.NotiDao
import com.example.test1.model.notify.Hit
import com.example.test1.model.notify.NotifyResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*
import kotlin.collections.ArrayList


class PostRepository()  {

    private var post = ArrayList<Data>()
    private val mutableLiveData = MutableLiveData<List<Data>>()
    private var dao: NotiDao? = null
//
    fun getMutableLiveData(): MutableLiveData<List<Data>> {

        val userDataService = APIClient.client
        val call = userDataService.post

        call?.enqueue(object : Callback<NotifyResponse?> {
            override fun onResponse(call: Call<NotifyResponse?>, response: Response<NotifyResponse?>) {
                val Post = response.body()
                if (Post != null && Post.hits!!.hits != null) {

                    post = convertData(Post.hits!!.hits) as ArrayList<Data>
                    mutableLiveData.value = post
                    Post.hits?.hits?.forEach {
                        val test = listOf<Hit>()
                        dao?.insert(test)
                    }

                }
            }
            override fun onFailure(call: Call<NotifyResponse?>, t: Throwable) {
                mutableLiveData.value=null
            }
        })
        return mutableLiveData
    }

}